src/Orleans.Transactions/ITransactionAgent.cs

```diff
@@ -37,5 +37,13 @@ public interface ITransactionAgent
         /// <returns>None.</returns>
         /// <remarks>This method is exception-free</remarks>
         Task Abort(TransactionInfo transactionInfo);
+
+        /// <summary>
+        /// Run transactional scope
+        /// </summary>
+        /// <param name="transactionOption"></param>
+        /// <param name="transactionScope"></param>
+        /// <returns></returns>
+        Task Transaction(TransactionOption transactionOption, Func<Task> transactionScope);
```

Comment:
Generally, this PR looks good to me. Where this method belongs is my only real question. A typical user would never need to use the other methods on `ITransactionAgent`. I wonder if the `Transaction` method should be moved, eg to a new type.

Reply:
I agree on that too. There are two new types introduced, the `ITransactionFrame` and `TransactionFrame` to expose the transactional scope. I didn't want the new type to be named `TransactionScope` since that can collide with the .NET type naming. Frame is kind of akin to scope though. My fantasy is limited...